test/routes/api/timetable.js

```diff
@@ -0,0 +1,100 @@
+    server.request.user = testUser;
+    chai
+      .request(server)
+      .post("/api/timetable/save")
+      .send(requestData)
+      .end(function (err, res) {
+        expect(res.status).to.be.equal(200);
+      });
+  });
+
+  it("/save does not allow empty requests", async function () {
+    server.request.isAuthenticated = function () {
+      return true;
+    };
+    testUser = {
+      name: "Why you wanna know it?",
+      email: "secret@secret.com",
+      branch: ["your head", "my foot"],
+      year: 1
+    };
+    server.request.user = testUser;
+    chai
+      .request(server)
+      .post("/api/timetable/save")
+      .end(function (err, res) {
+        expect(res.status).to.be.equal(400);
+      });
+  });
+
+  it("/save does not allow malformed requests 1", async function () {
+    server.request.isAuthenticated = function () {
+      return true;
+    };
+    testUser = {
+      name: "Why you wanna know it?",
+      email: "secret@secret.com",
+      branch: ["your head", "my foot"],
+      year: 1
+    };
+    requestData = { "dsfjndfd": "fdsfed" };
+    server.request.user = testUser;
+    chai
+      .request(server)
+      .post("/api/timetable/save")
+      .send(requestData)
+      .end(function (err, res) {
+        expect(res.status).to.be.equal(400);
+      });
+  });
+
+  it("/save does not allow malformed requests 2", async function () {
+    // server.request.isAuthenticated = function () {
+    //   return true;
+    // };
+    testUser = {
+      name: "Why you wanna know it?",
+      email: "secret@secret.com",
+      branch: ["your head", "my foot"],
+      year: 1
+    };
+    requestData = JSON.parse('{"id":null,"name":"timie timie"}');
+    server.request.user = testUser;
+    chai
+      .request(server)
+      .post("/api/timetable/save")
+      .send(requestData)
+      .end(function (err, res) {
+        console.log(res);
+        expect(res.status).to.be.equal(400);
```

Comment:
```suggestion
        expect(res.status).to.be.equal(400);
```

This will be 400 or 401? Auth middleware returns 401 on failed auth.

Reply:
400. This isn't for failed auth. This is for bad request

Reply:
Thats fine, but you commented out the force authentication code above, so even before it reaches the route, the middleware will return a 401 I think

Reply:
strange. the tests passed on my machine

Reply:
Nevermind, my bad, it keeps that config from the previous test.